src/github.com/matrix-org/dendrite/roomserver/input/latest_events.go

```diff
@@ -52,25 +52,12 @@ func updateLatestEvents(
 	if err != nil {
 		return
 	}
-	defer func() {
-		if err == nil {
-			// Commit if there wasn't an error.
-			// Set the returned err value if we encounter an error committing.
-			// This only works because err is a named return.
-			err = updater.Commit()
-		} else {
-			// Ignore any error we get rolling back since we don't want to
-			// clobber the current error
-			// TODO: log the error here.
-			updater.Rollback()
-		}
-	}()
 
 	u := latestEventsUpdater{
```

Comment:
I think the answer's no, but: is there any chance of this panicking (and thus wanting to be wrapped in the withTransaction call)?